bpf/libbpf/libbpf_api.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "libbpf.h"
+#include <linux/limits.h>
+#include <net/if.h>
+#include <bpf.h>
+#include <stdlib.h>
+#include <errno.h>
+
+static void set_errno(int ret) {
+	errno = ret >= 0 ? ret : -ret;
+}
+
+struct bpf_object* bpf_obj_open(char *filename) {
+	struct bpf_object *obj;
+	obj = bpf_object__open(filename);
+	set_errno(libbpf_get_error(obj));
+	return obj;
+}
+
+void bpf_obj_load(struct bpf_object *obj) {
+	set_errno(bpf_object__load(obj));
+}
+
+struct bpf_tc_opts bpf_tc_program_attach (struct bpf_object *obj, char *secName, int ifIndex, int isIngress) {
+	struct bpf_tc_opts opts;
+
+	DECLARE_LIBBPF_OPTS(bpf_tc_hook, hook, .attach_point = BPF_TC_EGRESS);
+	DECLARE_LIBBPF_OPTS(bpf_tc_opts, attach);
+
+	if (isIngress) {
+		hook.attach_point = BPF_TC_INGRESS;
+	}
+
+	attach.prog_fd = bpf_program__fd(bpf_object__find_program_by_name(obj, secName));
+	if (attach.prog_fd < 0) {
+		errno = -attach.prog_fd;
+		return opts;
```

Comment:
`opts` is uninitialised here.  Returning uninitialised memory can cause problems for CGO's validation logic.  Suggest zeroing it out before returning.